### PR TITLE
nixos/spice-webdavd: init 

### PR DESCRIPTION
**Pull request by kevincox**
###### Motivation for this change

Enable file sharing between a guest NixOS system and a host.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/phodav/default.nix [hunk 1]
```diff
@@ -21,7 +21,13 @@ in stdenv.mkDerivation rec {
...
   nativeBuildInputs = [ libsoup pkgconfig meson ninja ];
 
-  outputs = [ "out" "dev" "lib" ];
+  outputs = [ "out" "dev" "lib" "udev" ];
+
+  # We need to do this pre-configure otherwise the data/ folder disappears.
+  preConfigurePhases = ["udevInstallPhase"];
+  udevInstallPhase = ''
+    install -vDt $udev/lib/udev/rules.d/ data/*-spice-webdavd.rules
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/phodav/default.nix:30]: I don't think we need an extra output for this tiny file. Just put it into $out.

**Comment by kevincox** [comment]
It's been so long that I don't remember how I was testing this or whatever so I will just close it. It was working so anyone is welcome to pick it up.

**Pull request closed without merging**